src/state/slices/portfolioSlice/portfolioSlice.test.ts

```diff
@@ -19,456 +38,746 @@ import {
-        .toString(),
-      [yvusdcCaip19]: bn(tokenBalance(ethAccount1, yvusdcCaip19))
-        .plus(tokenBalance(ethAccount2, yvusdcCaip19))
-        .toString(),
-      [btcCaip19]: '1010'
-    },
-    ids: [ethCaip19, foxCaip19, usdcCaip19, yvusdcCaip19, btcCaip19]
-  },
-  accountBalances: {
-    byId: {
-      [ethAccountSpecifier1]: {
-        [ethCaip19]: '27803816548287370',
-        [foxCaip19]: '42729243327349401946',
-        [usdcCaip19]: '41208456',
-        [yvusdcCaip19]: '8178352'
-      },
-      [ethAccountSpecifier2]: {
-        [ethCaip19]: '23803816548287370',
-        [foxCaip19]: '40729243327349401946',
-        [usdcCaip19]: '41208456',
-        [yvusdcCaip19]: '8178352'
-      },
-      [btcAccountSpecifier]: {
-        [btcCaip19]: '1010'
-      }
-    },
-    ids: [ethAccountSpecifier1, ethAccountSpecifier2, btcAccountSpecifier]
-  },
-  accountSpecifiers: {
-    byId: {
-      [ethAccountSpecifier1]: [eth1Caip10],
-      [ethAccountSpecifier2]: [eth2Caip10],
-      [btcAccountSpecifier]: btcAccount.chainSpecific.addresses.map(
-        address => `${btcCaip2}:${address.pubkey}`
+  describe('selectPortfolioAssetAccounts', () => {
+    it('can get accounts containing an asset', () => {
+      const store = createStore()
+      const { ethAccount, ethAccount2 } = mockETHandBTCAccounts()
+
+      store.dispatch(
+        portfolioSlice.actions.upsertPortfolio(
+          accountToPortfolio({
+            [ethPubKeys[0]]: ethAccount,
+            [ethPubKeys[1]]: ethAccount2
+          })
+        )
       )
-    },
-    ids: [ethAccountSpecifier1, ethAccountSpecifier2, btcAccountSpecifier]
-  }
-}
 
-const state = {
-  ...mockStore,
-  assets: {
-    byId: {
-      [ethCaip19]: ethereum,
-      [foxCaip19]: fox
-    },
-    ids: [ethCaip19, foxCaip19]
-  },
-  marketData: {
-    ...mockStore.marketData,
-    byId: {
-      [ethCaip19]: {
-        price: '1000',
-        marketCap: '10000',
-        volume: '100000',
-        changePercent24Hr: 10
-      },
-      [foxCaip19]: {
-        price: '1',
-        marketCap: '10000',
-        volume: '100000',
-        changePercent24Hr: 10
-      }
-    },
-    ids: [ethCaip19]
-  },
-  portfolio: {
-    ...mockStore.portfolio,
-    assetBalances: {
-      byId: {
-        [ethCaip19]: '115607633096574740',
-        [foxCaip19]: '105458486654698803892'
-      },
-      ids: [ethCaip19, foxCaip19]
-    },
-    accountBalances: {
-      byId: {
-        [ethAccountSpecifier1]: {
-          [ethCaip19]: '27803816548287370',
-          [foxCaip19]: '42729243327349401946'
-        },
-        [ethAccountSpecifier2]: {
-          [ethCaip19]: '87803816548287370',
-          [foxCaip19]: '62729243327349401946'
-        }
-      },
-      ids: [ethAccountSpecifier1, ethAccountSpecifier2]
-    }
-  }
-}
+      const ethAccountId = `${ethAccount.caip2}:${toLower(ethAccount.pubkey)}`
+      const ethAccount2Id = `${ethAccount2.caip2}:${toLower(ethAccount2.pubkey)}`
```

Comment:
You're just testing this logic against the logic in the store. I'd prefer to test a hard-coded string value as the expected output. Otherwise you're testing test logic not a value.

This is done in a number of places. You can probably just do a search/replace for all of them.